crates/kitsune_p2p/dht_arc/src/dht_arc.rs

```diff
@@ -60,23 +63,143 @@ impl RangeBounds<u32> for ArcRange {
         u32: PartialOrd<U>,
         U: ?Sized + PartialOrd<u32>,
     {
-        unimplemented!("Contains doesn't make sense for this type of range due to redundant holding near the bounds. Use DhtArc::contains")
+        unimplemented!("Contains doesn't make sense for this type of range due to redundant holding near the bounds. Use DhtArcRange::contains")
+    }
+}
+
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    derive_more::Deref,
+    derive_more::DerefMut,
+    serde::Serialize,
+    serde::Deserialize,
+)]
+pub struct DhtArc(
+    #[deref]
+    #[deref_mut]
+    DhtArcRange,
+    Option<DhtLocation>,
+);
```

Comment:
sure? whatever you think : )

Reply:
Well I'm curious what you think too :) just seems that you pointed out an inconsistency due to the deref_mut, and if it weren't for that, the way it is now would be consistent.

Reply:
Honestly, I feel like we'd have less code to maintain, and less complexity if we only had one type and the understanding that the start location becomes arbitrary after set operations.

Reply:
For sure, it's a tradeoff. We'd have less types if we made everything Strings, too :) But I hear ya.

One thing to note is that in the quantized arc world, this distinction becomes even more significant, because then we have arcs with a definite starting DhtLocation, but actual start of the arc is somewhere *other* than that location in general (since it gets rounded down to the nearest chunk). And then we also have quantized arcs that are defined in terms of chunks, not DhtLocation. So it felt helpful to mirror that here. Maybe @freesig can chime in.